File: java/base/common/src/main/java/br/com/javanei/retrocenter/common/DatafileCatalogEnum.java
package br.com.javanei.retrocenter.common;

public enum DatafileCatalogEnum {
    NoIntro,
    TOSEC,
    MAME,
    HyperList,
    GoodSet;

    public static DatafileCatalogEnum fromName(String name) {
        if (name == null) {
            return null;
        }
        DatafileCatalogEnum r = parseName(name);
        if (r == null) {
            throw new IllegalArgumentException(name);
        }
        return r;
    }

    public static boolean isValid(String name) {
        return name != null && parseName(name) != null;
    }

    private static DatafileCatalogEnum parseName(String name) {
        if (name.toLowerCase().contains("no-intro") || name.toLowerCase().contains("nointro")) {
            return NoIntro;
        } else if (name.toUpperCase().contains("TOSEC")) {
            return TOSEC;
        } else if (name.toUpperCase().contains("MAME")) {
            return MAME;
        } else if (name.toLowerCase().equalsIgnoreCase("hyperlist")) {
            return HyperList;
        } else if (name.toLowerCase().equals("goodset")) {
            return GoodSet;
        }
        return null;
    }
}

File: net/Retrocenter/RetrocenterBase/br/com/javanei/retrocenter/common/DatafileCatalogEnum.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace RetrocenterBase.br.com.javanei.retrocenter.common
{
    enum DatafileCatalogEnum
    {
        NoIntro,
        TOSEC,
        MAME,
        HyperList,
        GoodSet
    }

    class DatafileCatalogEnumUtil
    {
        public static Nullable<DatafileCatalogEnum> FromName(string name)
        {
            if (name == null)
            {
                return null;
            }
            Nullable<DatafileCatalogEnum> r = ParseName(name);
            if (r == null || !r.HasValue)
            {
                throw new ArgumentException(name);
            }
            return r;
        }

        public static bool IsValid(string name)
        {
            return name != null && ParseName(name) != null;
        }

        private static Nullable<DatafileCatalogEnum> ParseName(string name)
        {
            if (name.ToLower().Contains("no-intro") || name.ToLower().Contains("nointro"))
            {
                return DatafileCatalogEnum.NoIntro;
            }
            else if (name.ToUpper().Contains("TOSEC"))
            {
                return DatafileCatalogEnum.TOSEC;
            }
            else if (name.ToUpper().Contains("MAME"))
            {
                return DatafileCatalogEnum.MAME;
            }
            else if (name.ToLower().Equals("hyperlist"))
            {
                return DatafileCatalogEnum.HyperList;
            }
            else if (name.ToLower().Equals("goodset"))
            {
                return DatafileCatalogEnum.GoodSet;
            }
            return null;
        }

    }
}
